### PR TITLE
TS3(TS-7?) - Create new customer via Manager Account

### PR DESCRIPTION
**Pull request by sharksystems**

### CODE REVIEW

File: src/main/java/pages/ManagerAddCustomerPage.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package pages;
+
+import com.codeborne.selenide.Condition;
+import org.openqa.selenium.By;
+
+import static com.codeborne.selenide.Selenide.$;
+
+public class ManagerAddCustomerPage extends BasePage<BasePage> {
```
Comment (samsometro) [inline src/main/java/pages/ManagerAddCustomerPage.java]:
```suggestion
public class ManagerAddCustomerPage extends BasePage<ManagerAddCustomerPage> {
```

File: src/main/java/pages/ManagerCustomersPage.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package pages;
+
+public class ManagerCustomersPage extends BasePage<BasePage>{
```
Comment (samsometro) [inline src/main/java/pages/ManagerCustomersPage.java]:
```suggestion
public class ManagerCustomersPage extends BasePage<ManagerCustomersPage>{
```

File: src/main/java/pages/ManagerOpenAccountPage.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    public static final By customerSelectField = By.xpath("//select[@id='userSelect']");
+    public static final By currencySelectField = By.xpath("//select[@id='currency']");
+    public static final By accountProcessBtn = By.xpath("//button[@type='submit' and contains(text(), 'Process')]");
+
+    public ManagerOpenAccountPage chooseCreatedCustomer (String name) {
+        $(customerSelectField).shouldBe(visible).selectOptionContainingText(name);
+        return this;
+    }
+    public ManagerOpenAccountPage chooseCustomerCurrency () {
+        $(currencySelectField).shouldBe(visible).selectOptionByValue("Rupee");
```
Comment (samsometro) [inline src/main/java/pages/ManagerOpenAccountPage.java:29]: create package data and create there enum Currency. Than call this enums in method "selectOptionByValue"

File: src/main/java/pages/ManagerAddCustomerPage.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    public ManagerAddCustomerPage enterCustomerLastName(String name){
+        $(lastNameField).shouldBe(Condition.enabled).sendKeys(name);
+        return this;
+    }
+    public ManagerAddCustomerPage enterRandomCustomerPostCode(String name){
+        $(postCodeField).shouldBe(Condition.enabled).sendKeys(name);
+        return this;
+    }
+    public ManagerAddCustomerPage clickCustomerSubmitBtn (){
+        $(addCustomerSubmitBtn).shouldBe(Condition.enabled).click();
```
Comment (samsometro) [inline src/main/java/pages/ManagerAddCustomerPage.java:38]: condition static import

File: src/test/java/tests/Test1.java [hunk 5]
```diff
@@ -1,20 +1,65 @@
...
+import static pages.HomePage.managerLoginBtn;
 import static pages.MainHeadPage.*;
+import static pages.ManagerMainPage.*;
 
 public class Test1 extends base.TestInit {
 
+    @Test
+    public void checkManagerCustomerCreation() {
+        //Create random credentials for user
+        String randomFirstName = faker.name().firstName();
```
Comment (samsometro) [inline src/test/java/tests/Test1.java]: No need in variables for faker

File: src/test/java/tests/Test1.java [hunk 6]
```diff
@@ -1,20 +1,65 @@
...
+        String randomLastName = faker.name().lastName();
+        String randomPostCode = faker.address().zipCode();
+
+        //Go to Manager Login > Add Customer and create a new user with random credentials
+        new HomePage().openManagerLoginBtn();
+        $(addCustomerBtn).shouldBe(enabled).click();
+        new ManagerAddCustomerPage()
+                .enterCustomerFirstName(randomFirstName)
```
Comment (samsometro) [inline src/test/java/tests/Test1.java:32]:
```suggestion
                .enterCustomerFirstName(random.name)
```

File: src/test/java/tests/Test1.java (hunk 5 above)
Comment (sharksystems) [reply src/test/java/tests/Test1.java]: Variables are used to locate the user we made within the test by his credentials

        By customerNameLocator = By.xpath("//td[contains(text(), '" + randomFirstName + "')]");

replacing them here with random.name would just create another random name

**Comment by sharksystems** [comment]
added new commit

**Pull request merged**